piksi_tools/ardupilot/mavlink_decode.py

```diff
@@ -0,0 +1,141 @@
+'''
+Takes in a dataflish BIN file and produces an SBP JSON log file with the following record fields:
+
+ {"delta": msec offset since beginning of run,
+  "timestamp": UTC timestamp (sec),
+  "data":  JSON representation of SBP message specialized to message type (like MsgBaselineNED),
+  "metadata": dictionary of tags, optional
+  }
+
+Requirements:
+
+  pip install pymavlink
+  sudo pip install sbp (see https://github.com/swift-nav/libsbp/tree/master/python/sbp for object definitions)
+
+'''
+from pymavlink.DFReader import *
+from construct import *
+from sbp.table import dispatch,  _SBP_TABLE
+from sbp.msg import SBP
+
+import time
+import binascii
+import json
+
+SBR1_DATASTART = 16
+SBR2_DATASTART = 13
+
+'''
+This function takes in a filename for a ArduPilot dataflash log,
+and returns an array of (timestamp, bytearray) tuples.
+The bytearray contains raw SBP binary data logged directly from the serial port.
+Each tuple should contain exactly one SBP message.
+
+This decoder requires commit 0b6e5ab1f6d7911d408aaee8a4ec7a457e238399
```

Comment:
I have a PR up on mavlink to allow you to get the binary directly without first parsing it and recreating.  It hasn't been commented on or merged, so for now you have to check out my branch or cherry pick the commit.  It might work without this, but it will be much harder.